### ISSUE TITLE
[BUG] Empty string saved instead of NULL

### ISSUE THREAD

**Issue by scrnjakovic** [phalcon/cphalcon#13550]
### Database
```sql
CREATE TABLE `accommodation` (
  ...
  `website` VARCHAR(60) DEFAULT NULL
  ...
```
### Controller
```php
// Request a transaction
$transaction = $this->transactionManager->get();

$accommodation = new Accommodation();
$accommodation->setTransaction($transaction);

...

$accommodationForm->bind(
    $_POST,
    $accommodation,
    [
        ...
        'website',
        ...
    ]
);

if ($accommodation->create() == false) {
    $transaction->rollback(__('Error has occured while creating accommodation'), $accommodation);
}

...

// no errors, commit changes
$transaction->commit();  
```
### Form
```php
// website input
$website = new Text('website');
$website->setLabel(__('Website'));
/*$website->addValidator(
    new Url(
        [
            "message" => __('Please enter valid website address')
        ]
    )
);*/
$this->add($website);
```
### Expected behavior

DB field should be `null` when nothing was entered in the field

### Actual behavior

DB field is empty string

Is this a bug or feature? I'll be bumped to hear it's a feature as it'd be hell to have to do following:
```php
public function setWebsite($website)
{
    if (empty($website)) {
        $this->website = null;
        
        return $this;
    }
    $this->website = $website;

    return $this;
}
```

### Details

* Phalcon version: 3.4.0
* PHP Version: 7.2.8
* Operating System: macOS
* Installation type: Compiling from source
* Server: Nginx
* Database: MySQL 8.0.12

**Comment by scrnjakovic** [comment]
Actually, this is more NFR as this is obviously due to the fact that all POST/GET values are strings, with the exception of array of course.

What I propose we do is have `Form::bind` turn empty strings into `null` by default before assigning it to the model. For those who, for whatever reason, want empty strings in their databases, they can allow it with something like

```php
$form->allowEmptyStrings(
    [
        'fieldName'
    ]
);
```
or
```php
$field = new Text(
    'field',
    [
        'allowEmptyString' => true
    ]
);
```

**Comment by niden** [comment]
@scrnjakovic Looking at this I have one concern. How would Phalcon know that an empty string is indeed an empty string vs. a null? There will be instances where the POST will be an empty string and this would be what I might want to add in the database vs a null.

Thoughs?

**Comment by scrnjakovic** [comment]
@niden Without getting into a discussion on why would one want empty strings in a database, I suppose there is a valid use case somewhere, I still think those cases are quite rare or at least less common than having `null`. So my idea was to treat all empty strings as nulls unless field is explicitly flagged to accept empty strings.

**Comment by zsilbi** [comment]
I am not a fan of storing empty strings either, but reasonable use cases max exist.
It's not an example I'd use personally, but in `view` you could decide to print the stored content if it's a `string` (even if it's empty), or use an alternative method to gather the content when it's `null`.

Filters can be used on `Form` elements, so I'd maybe write a `Filter` which returns null if the input is empty then set it to the elements I need.
If it's required on all the elements I'd override `Form::add()` in my base `Form`.

**Comment by niden** [comment]
@zsilbi I like this approach better because it extends the forms even more for developers.

We already have `Validators` for forms (valid/not valid). We can also attach Filters or Transformers or Decorators so that we can achieve this _transformation_

I will move this to the 4.1 release because this requires significant work.

@scrnjakovic I will add a new issue and close this one.

**Comment by scrnjakovic** [comment]
Yeah, I don't mind that approach, I just advocate NULL being default and empty string optional.
